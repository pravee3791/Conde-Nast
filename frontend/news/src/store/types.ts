import { Dispatch } from 'redux'


export type Thunk = (
  dispatch: Dispatch<any>,
  getState: () => RootState
) => Promise<any>

export type RootState = ReturnType<any>