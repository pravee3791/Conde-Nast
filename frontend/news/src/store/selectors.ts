
import { RootState } from './store'

export const news = (state: RootState) => state.News;