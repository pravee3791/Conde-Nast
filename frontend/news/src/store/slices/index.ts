import News from "./news";


const reducers = {
    News,
  
}
export default reducers