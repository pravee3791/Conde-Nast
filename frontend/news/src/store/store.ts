
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger'
import { Dispatch } from 'redux'
import appReducer from "./slices/index";


const middlewares = [logger]
const store = configureStore({
    reducer: appReducer,
    middleware: [...getDefaultMiddleware(), ...middlewares]
})

export default store;

// exporting types for TypeScript
export type RootState = ReturnType<typeof store.getState>;
export type Thunk = (
  dispatch: Dispatch<any>,
  getState: () => RootState
) => Promise<any>