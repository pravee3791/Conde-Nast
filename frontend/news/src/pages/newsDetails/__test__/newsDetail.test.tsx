
import NewsDetail from '../newsDetails';
import { shallow, mount } from 'enzyme';
import store from "../../../store/store";
import { Provider } from "react-redux";
import {
    BrowserRouter as Router,
} from "react-router-dom";

let wrapped = mount(<Provider store={store}> <Router><NewsDetail /></Router></Provider>);
describe('Testing the  Component', () => {
    test('Details component contains the heading', () => {
        expect(wrapped.find('h1').length).toEqual(1);
    })

    test('Details component contains the Search comp', () => {
        expect(wrapped.find(`img`).length).toEqual(1);
    })

    test('Details component contains the list of news', () => {
        expect(wrapped.find('section').length).toEqual(1);
    })

    test('Details component shall match the snapshot', () => {
        expect(wrapped).toMatchSnapshot();
    })

})

