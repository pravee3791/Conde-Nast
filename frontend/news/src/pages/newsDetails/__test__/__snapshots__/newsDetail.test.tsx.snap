// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the  Component Details component shall match the snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
   
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <NewsDetails>
        <Header>
          <div
            className="header"
          >
            <h4 />
          </div>
        </Header>
        <div
          className="newspage"
        >
          <h1
            className="title centerpage"
          />
          <img
            alt="sample88"
            src=""
          />
          <section>
            <p />
          </section>
          <div>
            <a
              href=""
              target="_blank"
            >
              Read More
            </a>
          </div>
        </div>
      </NewsDetails>
    </Router>
  </BrowserRouter>
</Provider>
`;
